bots/historybot/history.bot.js

```diff
@@ -12,7 +12,18 @@ const wait = async (n,val)=>new Promise((r)=>setTimeout(()=>r(val), n));
 
 let HistoryStorage = {};
 
-const getDate = (ts)=>datefns.format(new Date(ts*1000), 'YYYY-MM-DD H:mm:ss')
+const getDate = (ts)=>datefns.format(new Date(ts*1000), 'YYYY-MM-DD H:mm:ss');
+
+const uploadToDatabase = (endpoint, payload)=>{
+	const HistoryDatabaseToken = config.get('historybot.db_token');
+	if (!HistoryDatabaseToken) return;
+
+	const host = config.get('historybot.db_host');
+	return request.post(`https://${host}/${endpoint}`)
```

Comment:
Since this happens on every message, I could see some potential for storing these config values and the URL in variables outside the function scope. Not a big deal, though, as that would make the code a bit more scattered.

Reply:
I like this. Try to assemble the endpoint at the top of the file at runtime, maybe even throw a warning if the token isn't set.

Reply:
I'm happy to move the config getters out to the top of the file, but I'm not sure the rest is a good idea.

There are currently 2 endpoints, and eventually there are going to be _at least_ 3.  It feels clunky to maintain a list of available endpoints at the top of the file, requiring extra code changes when adding an endpoint, instead of just assembling the endpoint at calltime so we can call this function with a new endpoint and have it just work.

Re: the token - Right now I'm just automatically disabling the upload if the token isn't set, which I think is nice for dev environments so you're not forced to have a dependency on this feature when you're implementing something else, and don't get warnings spammed at you every time you start up your local.  But if you disagree, I'm happy to change it.